### DISCUSSION TITLE
concurrent.futures and limit before errors

### DISCUSSION THREAD

**Discussion by joaosousa997** [ktbyers/netmiko#3126] [Q&A]
i am using 75, after that it gives me an EOL Error and a read banner error as well, i think it get's really unstable.
Has anyone tried using more than 75 threads?

```
from concurrent.futures import ThreadPoolExecutor
try:
        max_threads = 75 #max 50
        with ThreadPoolExecutor(max_workers=max_threads) as executor:
```

For context i hop to a jump server via ConnectHandler() then i write_channel() ssh to the router i want, i need to do this for 7k routers.

If anyone knows a faster way than concurrent.futures i would like to try it out!

**Comment by ktbyers** [comment]
jump_server is likely going to cause issues as you are creating separate processes at the system level for this.

But yes have definitely seen people use over 75 threads.

In your case, I would be worried the Jump Server setup is going to be a limiting factor.

**Reply by ktbyers** [reply]
Also your AAA setup can be a limiting factor i.e. AAA can be very limiting on how many simultaneous logins it can handle (and if you are doing command authorization that can also be a big issue).
